OWML.ModHelper/ModConfig.cs

```diff
@@ -26,16 +42,28 @@ public T GetSettingsValue<T>(string key)
                 return default;
             }
 
-            var value = Settings[key];
+            var setting = Settings[key];
+            var type = typeof(T);
 
             try
             {
-                var val = value is JObject obj ? obj["value"] : value;
-                return (T)Convert.ChangeType(val, typeof(T));
+                var value = setting is JObject objectValue ? objectValue["value"] : setting;
+
+                if (type.IsEnum && value is float || value is double)
+                {
+                    var floatValue = Convert.ToDouble(value);
+                    value = (long)Math.Round(floatValue);
+                }
+
+                if (!type.IsEnum || value is int || value is long)
```

Comment:
with the method extraction it would be better to un-invert the if again ;-P

Reply:
no longer valid :P